### ISSUE TITLE
WebDAV: Detect if source and destination end up pointing to the same folder (loop)

### ISSUE THREAD

**Issue by francoisWeber** [photoprism/photoprism#3088]
Hi, 

I installed photoprism through the `docker-compose.yaml` file provided by your team (`photoprism/photoprism`, image_id: `e6f531bbf1b0`). I'm running a Nextcloud (NC) instance on the same device and NC stores my pictures in `/var/nextcloud/data/francois/files/Pictures`. Thus, in my DockerCompose configuration I set
```yaml
    [...]
    environment:
      PHOTOPRISM_READONLY: "true"
    volumes:
      - "/var/nextcloud/data/francois/files/Pictures:/photoprism/originals"
    [...]
```

At the beginning, everything worked fine. Then I activated the WebDAV sync option from Photoprism interface, connecting my NC account through webdav (I wondered the use-case of this option). 
A few minutes later, I noticed a huge sync activity on my `/Pictures` folder on my NC account : thousands of pictures changed server-side, which was not supposed to happen. I checked the freshly synchronized pictures and it appears most of them were broken versions of the original data (don't worry I have a backup). Here is an example showing the size decay:
```
-rwx------  1 f.weber  staff   2,1M 28 jui  2020 20-06-28 20-22-54 7551-original.jpg*
-rw-r--r--  1 f.weber  staff   384K  5 jan 12:54 20-06-28 20-22-54 7551.jpg
```
Both images, original one and broken one can be found here: https://cloud.fweber.fr/index.php/s/aNitf3KFiZ3dwAz

Here is the line of Photoprism log that deals with this sync:
```
time="2023-01-05T11:54:20Z" level=info msg="sync: uploaded 'Amis - Famille/Bilingues/Photos Rouxonnière/20-06-28 20-22-54 7551.jpg' to '/Pictures/Amis - Famille/Bilingues/Photos Rouxonnière/20-06-28 20-22-54 7551.jpg' (Cloud)"
```

I think it's a really annoying bug because syncing broken pictures can be really dangerous for those who have no backup.
Hope this will help !

**Comment by graciousgrey** [comment]
Thank you for reporting this!

> At the beginning everything worked fine. Then I enabled the WebDAV sync option on the Photoprism interface and connected my NC account via WebDAV (I was wondering what this option was for).

Could you briefly explain your goal(s)? If you have configured your Nextcloud photo folder as PhotoPrism's Originals folder, the contents will always be the same, so there is no need to sync.

The [Sync](https://docs.photoprism.app/user-guide/sync/services-sync/) is meant to upload or download files to other folders, such as other PhotoPrism instances or Nextcloud instances that do not access the same folder.

We will try to reproduce this issue and see if we can prevent the sync from starting if the destination folder is the same as the source folder.

Also, please let us know if any part of our documentation was misleading in this regard and how we can improve it if so.

**Comment by francoisWeber** [comment]
Actually I had no specific goal when I tried to plug back my WebDAV into Photoprism, just curiosity. The French documentation maybe lacks a bit of accuracy concerning the "loading" option which was the one I wanted to experiment.

Nonetheless, I made this broken-JPG bug appears and I'm not sure it's related to the "double sync" weird stuff I did by plugging NC twice into Photoprism.

Thank you for this quick answer !!
François

**Comment by graciousgrey** [comment]
Thank you very much for your feedback!

> The French documentation maybe lacks a bit of accuracy concerning the "loading" option which was the one I wanted to experiment.

What french documentation are you referring to? The french translation of the Web-UI?

>Nonetheless, I made this broken-JPG bug appears and I'm not sure it's related to the "double sync" weird stuff I did by plugging NC twice into Photoprism.

We will try to find out if the double sync caused the broken JPGs or if there might be some other cause.

**Comment by francoisWeber** [comment]
> What french documentation are you referring to? The french translation of the Web-UI?

yes, actually [this part of the doc](https://docs.photoprism.app/user-guide/settings/sync/#edit-sync-settings) showing two sync options "Download remote files" vs "upload local files" was not really clear to me (in French but also in English actually 😆)

> We will try to find out if the double sync caused the broken JPGs or if there might be some other cause.

🤞 Thanks !

**Comment by lastzero** [comment]
But now it is clear? "Upload local files" means that you upload files to the remote server, e.g. to Nextcloud, possibly overwriting files that already exist there. "Download remote files" means that you download files e.g. from Nextcloud, possibly overwriting local files with the same name, especially if "Preserve filenames" is selected.

**Comment by francoisWeber** [comment]
Now it's clear but maybe a very light explanation below each option could be helpful ? I must admit I was a bit confused despite my tech background 😇

**Comment by dvpfig** [comment]
Is it really possible for PhotoPrism to detect if source and destination are pointing to the same folder? I mean the paths are different from PhotoPrism perspective and the fact that the list of files is the same doesn't mean it's the same instance (it could have been two different instances of nextcloud). 

One of the folders is a local folder (that by chance is the nextcloud folder of this user), how would PhotoPrism know that a folder added via the GUI via the service Nextcloud points to the same folder as the local one? 

Maybe just a warning in the documentaiton for someone to avoid falling in this mistake. But probably not possible to detect it in PhotoPrism.
